web/src/beta/features/Dashboard/ContentsContainer/Projects/hooks.ts

```diff
@@ -0,0 +1,143 @@
+  const navigate = useNavigate();
+  const state = localStorage.getItem("viewState");
+  const [selectedProjectId, setSelectedProjectId] = useState<string | undefined>(undefined);
+  const [visible, setVisible] = useState(false);
+  const [viewState, setViewState] = useState(state ? state : "grid");
+  const [isStarred, setIsStarred] = useState<Record<string, boolean>>({});
+
+  const handleProjectCreate = useCallback(
+    async (data: Pick<Project, "name" | "description" | "imageUrl">) => {
+      if (!workspaceId) return;
+      await useCreateProject(
+        workspaceId,
+        Visualizer.Cesium,
+        data.name,
+        true,
+        data.description,
+        data.imageUrl || "",
+      );
+    },
+    [useCreateProject, workspaceId],
+  );
+
+  const handleProjectUpdate = useCallback(
+    async (project: Project, projectId: string) => {
+      await useUpdateProject({ projectId, ...project });
+    },
+    [useUpdateProject],
+  );
+
+  const {
+    projects: projectsData,
+    loading,
+    fetchMore,
+    networkStatus,
+  } = useProjectsQuery(workspaceId);
+
+  const projectNodes = projectsData?.edges.map(e => e.node);
+
+  const projects = useMemo(() => {
+    return (projectNodes ?? [])
+      .map<Project | undefined>(project =>
+        project
+          ? {
+              id: project.id,
+              description: project.description,
+              name: project.name,
+              imageUrl: project.imageUrl,
+              isArchived: project.isArchived,
+              sceneId: project.scene?.id,
+              updatedAt: new Date(project.updatedAt),
+            }
+          : undefined,
+      )
+      .filter((project): project is Project => !!project);
+  }, [projectNodes]);
+
+  const hasMoreProjects =
+    projectsData?.pageInfo?.hasNextPage || projectsData?.pageInfo?.hasPreviousPage;
+
+  const isRefetchingProjects = networkStatus === 3;
+
+  const handleGetMoreProjects = useCallback(() => {
+    if (hasMoreProjects) {
+      fetchMore({
+        variables: {
+          before: projectsData?.pageInfo?.endCursor,
+        },
+        updateQuery: (prev, { fetchMoreResult }) => {
+          if (!fetchMoreResult) return prev;
+          return fetchMoreResult;
+        },
+      });
+    }
+  }, [hasMoreProjects, fetchMore, projectsData?.pageInfo?.endCursor]);
+
+  const selectedProject = useMemo(
+    () => projects.find(project => project.id === selectedProjectId) || undefined,
+    [projects, selectedProjectId],
+  );
+
+  const handleProjectSelect = useCallback(
+    (projectId: string | undefined) =>
+      setSelectedProjectId(prevId =>
+        prevId === projectId || projectId === undefined ? undefined : projectId,
+      ),
+    [],
+  );
+
+  const handleVisibility = useCallback(() => {
+    setVisible(!visible);
+  }, [visible]);
+
+  const handleViewChange = (newView: "grid" | "list") => {
+    localStorage.setItem("viewState", newView);
```

Comment:
I think it's better to define a constant and export it for this storage key, using the same one when get/set it. The naming can be more specific for example `reearth-visualizer-dashboard-projects-view-state`